### ISSUE TITLE
Bowtie2 command line options

### ISSUE THREAD

**Issue by jakublipinski** [calico/basenji#88]
Thanks for creating this incredible tool. 

I'd like to rebuild your data from the FASTA files to try some different data processing. In the paper you mention:

> We aligned the reads with Bowtie 2, requesting the program return a maximum of 10 multimapping alignments

What are the exact command line arguments you used with `Bowtie2` to create a `BAM` file?

Alternatively, are your preprocessed *.w5 files (e.g. `/home/drk/tillage/datasets/human/cage/fantom/CNhs13797/summary/coverage.w5`) available for download? 

Thanks!

**Comment by davek44** [comment]
In the more recent processing for this paper (https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008050), I switched to BWA which has the nice property that all multi-mapping alignments are specified on one row.

My parameters for single end CAGE is the following:
bwa mem -k 19 -w 100 -r 1.5 -c 500 -h 16 -T 28 -M 

Then I create coverage tracks for CAGE using:
bam_cov.py -g -m 4 -s 4 --smooth_out 2 --clip_max 32  --clip_multi 8

I have the .w5 files on disk, but they make up several terabytes of data. I could make a subset available to you, but offering the whole thing would be challenging.

**Comment by cristianregep** [comment]
What is smooth_out? It doesn't seem to be an option for bam_cov.py . 

Also relevant to this thread, where did you get fastq files for CAGE? There only seem to be bam files on the FANTOM5 repository at the current time https://fantom.gsc.riken.jp/5/datafiles/latest/basic/human.primary_cell.hCAGE/ ?

Great work with Basenji and Enformer!

**Comment by davek44** [comment]
smooth_out is specified here: https://github.com/calico/basenji/blob/master/bin/bam_cov.py#L92
It is then applied here to smooth the final coverage before writing to disk: https://github.com/calico/basenji/blob/master/bin/bam_cov.py#L1579

The FANTOM FASTQ files are available on their FTP site. Here's an example URL: ftp://ftp.ddbj.nig.ac.jp/ddbj_database/dra/fastq/DRA002/DRA002748/DRX023625/DRR026020.fastq.bz2

**Comment by sb-vsfishman** [comment]
Thanks for these comments! 
I have a question whether `bam_cov.py` includes the read depth normalization? I have reprocessed several CAGE .bam files using this script recently, and found that in the produced .bigWig files mean value of 200-bp bins vary at leas 10 times.

**Comment by davek44** [comment]
No, the script doesn’t perform read depth normalization. I use a Poisson loss for the predictions so I want to maintain the depth information. But I doubt it would hurt to normalize.

**Comment by sb-vsfishman** [comment]
I suppose, that if I insert 
`chrom_coverage_array = chrom_coverage_array / np.sum(chrom_coverage_array)` 
at https://github.com/calico/basenji/blob/47cb8ecf0f33ec98d096080ba52f8034149a393f/bin/bam_cov.py#L1596

it will do the normalization job...

**Comment by davek44** [comment]
Yup, that looks roughly correct to my eye. Be careful about types though because summing chrom_coverage_array might overflow. Also, it's typical to scale to something like 10k or 1million.

**Comment by liqingbioinfo** [comment]
Hi Dave,
     
     Your explanations are really informative. I wonder if I can apply the same BWA command from your CAGE data (bwa mem -k 19 -w 100 -r 1.5 -c 500 -h 16 -T 28 -M) to my ChIP-Seq data? If not, do you have any suggestions for parameters used in BWA? 
     I really appreciate your help, and I look forward to hearing from you. 

Yours sincerely 
Leah

**Comment by davek44** [comment]
Yes, I think similar parameters should work fine for ChIP-seq

**Comment by liqingbioinfo** [comment]
Great! Thanks a million.
